### PR TITLE
Split FieldValidator.getInvalidityInfo() into validation methods for specific fields #5283

### LINKED ISSUE #5283: Split FieldValidator.getInvalidityInfo() into validation methods for specific fields
Method to be refactored is: [`getInvalidityInfo(FieldType fieldType, String fieldName, Object value)`][link]

It currently decides which specific validation method to call by switching over a `FieldType` argument passed in by the client code.

This actually means that the client code already knows which specific validation is needed, and the decision making in `getInvalidityInfo()` is rather redundant.

e.g., `AccountAttributes` currently calls `validator.getInvalidityInfo(FieldValidator.FieldType.PERSON_NAME, name)`, it can be changed to `validator.getInvalidityInfoForPersonName(name)`

[link]: https://github.com/TEAMMATES/teammates/blob/master/src/main/java/teammates/common/util/FieldValidator.java#L431

### PR DESCRIPTION
**Pull request by YongJieYongJie**
Fixes #5283

### CODE REVIEW

**Comment by YongJieYongJie** [comment]
// currently extracted one case, 15 remains

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 1]
```diff
@@ -498,6 +494,10 @@ public String getInvalidityInfo(FieldType fieldType, String fieldName, Object va
             return returnValue;
         }
     }
+
+    public String getInvalidityInfoForPersonName(String personName) {
```
Comment (damithc) [inline src/main/java/teammates/common/util/FieldValidator.java]: Add header comments.

**Comment by JosephineKwa** [comment]
Hey @YongJieYongJie will you be doing the other cases in this PR too, or in separate PRs?

**Comment by YongJieYongJie** [comment]
@JosephineKwa Thanks for reminding, I'll split them into smaller issues. So for now, just this 9 extraction.

**Comment by JosephineKwa** [comment]
Do notify me when this pr is ready for review :)

**Comment by YongJieYongJie** [comment]
@JosephineKwa Ready for review!

File: src/test/java/teammates/test/cases/common/FieldValidatorTest.java [hunk 2]
```diff
@@ -254,98 +254,143 @@ public void testGetValidityInfoForAllowedName() {
...
-        
-        //See NOTE 1.
-        runGenericTestCasesForCappedSizeStringTypeField(
-                FieldType.INSTITUTE_NAME, 
-                INSTITUTE_NAME_MAX_LENGTH, 
-                INSTITUTE_NAME_ERROR_MESSAGE, 
-                false);
+
+    private void testGetInvalidityInfo_PersonName() {
+        invalidityInfoFor_validName_shouldBeEmptyString();
```
Comment (JosephineKwa) [inline src/test/java/teammates/test/cases/common/FieldValidatorTest.java]: Should this be "shouldReturnEmptyString" instead?

**Comment by YongJieYongJie** [comment]
@JosephineKwa Ready for review!

I renamed the `shouldBe` to `shouldReturn`
I also wrapped some long lines

**Comment by JosephineKwa** [comment]
Sorry! Created merge conflicts again after merging in your other PR :disappointed:

**Comment by YongJieYongJie** [comment]
@JosephineKwa Conflicts merged!

**Pull request merged**